namer/namer.cc

```diff
@@ -1852,6 +1859,40 @@ class TreeSymbolizer {
                     }
                 }
 
+                if (send->block() != nullptr) {
+                    if (const auto *hash = ast::cast_tree<ast::Hash>(send->block()->body)) {
+                        for (const auto &keyExpr : hash->keys) {
+                            const auto *key = ast::cast_tree<ast::Literal>(keyExpr);
+                            if (key == nullptr || !key->isSymbol(ctx)) {
+                                if (auto e =
+                                        ctx.beginError(keyExpr.loc(), core::errors::Namer::InvalidTypeDefinition)) {
+                                    e.setHeader("Hash provided in block to `{}` must have symbol keys",
+                                                send->fun.show(ctx));
+                                }
+                                return tree;
+                            }
+
+                            switch (key->asSymbol(ctx).rawId()) {
+                                case core::Names::fixed().rawId():
+                                    fixed = true;
+                                    break;
+
+                                case core::Names::lower().rawId():
+                                case core::Names::upper().rawId():
+                                    bounded = true;
+                                    break;
+                            }
```

Comment:
I think we should error on unknown keys, though I don't know whether it's better to do it here or in type syntax parsing.